### PR TITLE
Always check password quality

### PR DESCRIPTION
**Pull request by rra**
Previous versions bypassed the password quality check in `kadmind` and `kpasswdd` if the the password was being changed by an administrator instead of the user.  However, password quality checks are how password history should be implemented, and administrator password changes should also go into history.  Check password quality unconditionally.

### CODE REVIEW

File: kadmin/server.c [hunk 1]
```diff
@@ -329,6 +330,12 @@ kadmind_dispatch(void *kadm_handlep, krb5_boolean initial,
 	    && krb5_principal_compare (contextp->context, contextp->caller,
 				       princ))
 	{
+	    ret = 0;
```
Comment (nicowilliams) [inline kadmin/server.c]: Wouldn't it be better to just reverse the sense of the if condition and then drop this `ret = 0; } else {` bit?

**Comment by macrotex** [comment]
The changes in test #293 were written to address this exact patch. When you say "This definitely needs to be bundled with #293" do you mean that #320 should be altered to include the patch from #293, and then #293 deleted?

**Comment by jaltman** [comment]
I believe that always checking password quality should be a configuration option.

**Comment by lhoward** [comment]
It's interesting (but I suppose there may be historical reasons for this) that `allow_self_change_password` is honored only by kadmind (not kpasswdd).

**Comment by lhoward** [comment]
Anyway, I'll integrate with a configuration knob, default to existing behavior?

**Comment by lhoward** [comment]
Someone come up with a better name than `check_on_admin_set` for the option :)

**Comment by lhoward** [comment]
Closing, will resubmit new pull request.

**Comment by nicowilliams** [comment]
> It's interesting (but I suppose there may be historical reasons for this) that allow_self_change_password is honored only by kadmind (not kpasswdd).

I think the reason is just that users were supposed to use kpasswdd for password changes.

**Pull request closed without merging**